src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

```diff
@@ -70,18 +71,39 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 	 * Called when the drag operation starts.
 	 */
 	public onDragStart(tree: ITree, dragAndDropData: IDragAndDropData, originalEvent: DragMouseEvent): void {
+		let escapedSchema, escapedName, finalString;
 		TreeUpdateUtils.isInDragAndDrop = true;
 		const data = dragAndDropData.getData();
 		const element = data[0];
-		if (element.nodeTypeId === 'Column' || element.nodeTypeId === 'Table') {
-			const escapedSchema = element.metadata.schema?.replace(/]/g, ']]');
-			const escapedName = element.metadata.name?.replace(/]/g, ']]');
-			const finalString = escapedSchema ? `[${escapedSchema}].[${escapedName}]` : `[${escapedName}]`;
+		if (element.nodeTypeId === 'Column' || element.nodeTypeId === 'Table' || element.nodeTypeId === 'View') {
+			escapedSchema = this.dragDropHelper(element.metadata.schema);
+			escapedName = this.dragDropHelper(element.metadata.name);
+			finalString = escapedSchema ? `[${escapedSchema}].[${escapedName}]` : `[${escapedName}]`;
 			originalEvent.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([`${element.nodeTypeId}:${element.id}?${finalString}`]));
 		}
+		if (element.nodeTypeId === 'Folder' && element.label === 'Columns') {
+			// get children
+			let returnString = '';
+			for (let child of element.children) {
+				escapedSchema = this.dragDropHelper(child.metadata.schema);
+				escapedName = this.dragDropHelper(child.metadata.name);
+				finalString = escapedSchema ? `[${escapedSchema}].[${escapedName}]` : `[${escapedName}]`;
+				returnString = returnString ? `${returnString},${finalString}` : `${finalString}`;
+			}
+
+			originalEvent.dataTransfer.setData(DataTransfers.RESOURCES, JSON.stringify([`${element.nodeTypeId}:${element.id}?${returnString}`]));
+		}
 		return;
 	}
 
+	private dragDropHelper(input: string): string {
```

Comment:
1. The function name isn't really descriptive of what's actually happening, would suggest renaming to something like `escapeString`
2. Should be returning undefined, not null
3. Return type should be `string | undefined`